src/test_l1_dsFPD.c

```diff
@@ -744,86 +727,86 @@ void test_l1_dsFPD_negative_dsSetFPBrightness (void)
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     dsError_t result;
     dsFPDIndicator_t eIndicator;
-    int count = 0;
+    uint8_t count = 0;
     char buffer[DS_FPD_KEY_SIZE];
     char minbuffer[DS_FPD_KEY_SIZE], maxbuffer[DS_FPD_KEY_SIZE];
     dsFPDBrightness_t minBrightness;
     dsFPDBrightness_t maxBrightness;
 
-
-    // Variation 01: Call dsSetFPBrightness() without calling dsFPInit()
-    count = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
+    // Step 01: Call dsSetFPBrightness() without calling dsFPInit()
+    count = UT_KVP_PROFILE_GET_UINT8("dsFPD/Number_of_Indicators");
     for(int i = 1; i <= count; i++)
     {
 	snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
-        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
         snprintf(minbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MIN_BRIGHTNESS",eIndicator);
-        minBrightness =  UT_KVP_PROFILE_GET_UINT32(minbuffer);
+        minBrightness =(dsFPDBrightness_t)UT_KVP_PROFILE_GET_UINT32(minbuffer);
         snprintf(maxbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MAX_BRIGHTNESS",eIndicator);
-        maxBrightness = UT_KVP_PROFILE_GET_UINT32(maxbuffer);
-        int avgBrightness = (maxBrightness + minBrightness) / 2;
+        maxBrightness =(dsFPDBrightness_t)UT_KVP_PROFILE_GET_UINT32(maxbuffer);
+        uint32_t avgBrightness = (maxBrightness + minBrightness) / 2;
         result = dsSetFPBrightness(eIndicator, avgBrightness);
         CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_NONE);
     }
 
-    // Variation 02: Initialize with dsFPInit()
+    // Step 02: Initialize with dsFPInit()
     result = dsFPInit();
     UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
-    // Variation 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
-    // Variation 04: Pass an invalid eBrightness parameter to dsSetFPBrightness()
+    // Step 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
+    // Step 04: Pass an invalid eBrightness parameter to dsSetFPBrightness()
     for(int i = 1; i <= count; i++)
     {
 	snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
-        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
         UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, eIndicator);
         result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
         result = dsSetFPBrightness(eIndicator, 200);
```

Comment:
Instead of 200, can we give maxBrightness (dsFPD_BRIGHTNESS_MAX) + 1 here.  That will make sense rather than giving a too far value.